contracts/JBSplitsStore.sol

```diff
@@ -183,10 +208,77 @@ contract JBSplitsStore is IJBSplitsStore, JBOperatable {
         revert INVALID_TOTAL_PERCENT();
       }
 
-      // Push the new split into the project's list of splits.
-      _splitsOf[_projectId][_domain][_group].push(_splits[_i]);
+      uint256 _packedSplitParts1 = _splits[_i].preferClaimed ? 1 : 0;
+      _packedSplitParts1 |= _splits[_i].percent << 1;
+      _packedSplitParts1 |= _splits[_i].projectId << 33;
```

Comment:
an upper bounds check of the projectId is also good to have

Reply:
max split percentage is 1000000000 so that would be 30 bits; shouldn't we shift by 31 here?

Reply:
true. we originally had it in uint32, which is a cap of 4,294,967,296. I reduced it to 1,000,000,000 but didnt realize this would fit in fewer bits. 

Doesn't hurt to keep it in 32, we dont need the extra bits.

Reply:
upper bound check on projectId is good